src/data-types/connect.c

```diff
@@ -244,13 +251,15 @@ int mail_tcp_connect_with_local_address_timeout(const char * server, uint16_t po
       continue;
     
     // Christopher Lyon Anderson - prevent SigPipe
+    // patch by fdik
+if (!libetpan_deliver_sigpipe) {
```

Comment:
On Thu, Mar 05, 2015 at 01:36:02PM -0800, Hoà V. DINH wrote:
> > @@ -244,13 +251,15 @@ int mail_tcp_connect_with_local_address_timeout(const char * server, uint16_t po
> >        continue;
> >      
> >      // Christopher Lyon Anderson - prevent SigPipe
> > +    // patch by fdik
> > +if (!libetpan_deliver_sigpipe) {
> Do we need to customize that behavior?

Yes. That's what it's all about. When a timeout occurs, we need a
SIGPIPE. Because of compatibility reasons, this has to be switched on.

Having the wrappers is the first step. Making it configurable when I/O
shoudln't fail will be the second one.

Yours,
VB.
-- 
Volker Birk
Oberer Graben 4, 8400 Winterthur, Schweiz
mailto:vb@dingens.org  http://fdik.org